### ISSUE TITLE
0.1.22 Fixes to support Vagrant 2.4.3

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Super.Human.Installer#126]
@PanProminic and @piotrzarzycki21 reported some errors with Vagrant 2.4.3:

>1) First was:
>
>Path: /Users/piotrzarzycki/Library/Application Support/SuperHumanInstallerDev/servers/demo-tasks/6530/core/Hosts.rb 
>Line number: 0 
>Message: undefined method `exists?'
>
>I resolved it by changing that method to 'exist' - exists seems to be deprecated with newest Ruby. After my changes I got: 
>
>2) 
>
>There was an error loading a Vagrantfile. The file being loaded 
>and the error message are shown below. This is usually caused by 
>a syntax error. 
>
>Path: <provider config: virtualbox> 
>Line number: 175 
>Message: NoMethodError: undefined method `=~' for an instance of Integer 
>[SHI]: 'vagrant up' stopped with exit code: 1, elapsed time: 00:00:01
>
>Issue is in Hosts.rb file in line 175 - I did tweak that part and it's working for me, but it is rather workaround. I just wanted to point you and bring this to your attention, because some of us may want to upgrade stuff on their system and bump into these issues. 

I think we can resolve these problems in this file:  https://github.com/Moonshine-IDE/Super.Human.Installer/blob/master/Assets/provisioners/demo-tasks/0.1.22/scripts/core/Hosts.rb

Note that this is already fixed in the 0.1.23 provisioners, which are not merged yet (#118).  However, the 0.1.22 provisioners will still be available in the UI unless we remove them.

**Comment by JoelProminic** [comment]
I did a test with 0.1.22 today, and I was reminded that this provisioner will always fail with:

```
TASK [startcloud_sdkman_install : Update SDKMAN] *******************************
[WARNING]: Skipping plugin (/usr/lib/python3/dist-
packages/ansible/plugins/filter/core.py), cannot load: cannot import name
'_is_unsafe' from 'ansible.utils.unsafe_proxy' (/usr/lib/python3/dist-
packages/ansible/utils/unsafe_proxy.py)
fatal: [3029--test-jedi-22.shi.com]: FAILED! => 
  msg: |-
    The conditional check 'sdkman_update | bool' failed. The error was: template error while templating string: Could not load "bool": "cannot import name '_is_unsafe' from 'ansible.utils.unsafe_proxy' (/usr/lib/python3/dist-packages/ansible/utils/unsafe_proxy.py)". String: {% if sdkman_update | bool %} True {% else %} False {% endif %}. Could not load "bool": "cannot import name '_is_unsafe' from 'ansible.utils.unsafe_proxy' (/usr/lib/python3/dist-packages/ansible/utils/unsafe_proxy.py)"
  
    The error appears to be in '/vagrant/ansible/roles/startcloud_sdkman_install/tasks/sdkman.yml': line 21, column 3, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
    -
      name: Update SDKMAN
      ^ here
[3029--test-jedi-22.shi.com] TASK: startcloud_sdkman_install : Update SDKMAN (debug)> Use exit() or Ctrl-D (i.e. EOF) to exit
[3029--test-jedi-22.shi.com] TASK: startcloud_sdkman_install : Update SDKMAN (debug)> User interrupted execution
Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
[SHI]: 'vagrant up' stopped with exit code: 1, elapsed time: 00:07:52
```

My workaround was to comment out or remove these tasks:
https://github.com/Moonshine-IDE/Super.Human.Installer/blob/dfaa1baeb753e0e4d86fadb675fc584a6a710ad4/Assets/provisioners/demo-tasks/0.1.22/scripts/provisioners/ansible/roles/startcloud_sdkman_install/tasks/sdkman.yml#L21-L27
https://github.com/Moonshine-IDE/Super.Human.Installer/blob/dfaa1baeb753e0e4d86fadb675fc584a6a710ad4/Assets/provisioners/demo-tasks/0.1.22/scripts/provisioners/ansible/roles/startcloud_sdkman_install/tasks/sdkman.yml#L67-L79

@MarkProminic made a better solution for this for 0.1.23, but it may not be worth integrating it if it is complex.

I thought I had an issue with more details for this.   I'll link it if I find it.

**Comment by MarkProminic** [comment]
If you change the vagrant box template to anything higher than 0.0.6 (at time of writing 0.0.9) it should get around this issue.

The problem is that, while Ansible is running, it runs the dependency role(https://github.com/STARTcloud/startcloud_roles/tree/main/roles/dependencies), which does an:

```
apt update
```

and

```
apt upgrade
```


Because of these OS package updates (which is included for security reasons), Ansible itself is updated WHILE Ansible is running, for the most part Ansible remained the same, however when it handles specifc tasks during the SDKMAN role, it fails because of changes to the Ansible package by the maintainers.

**Comment by JoelProminic** [comment]
@MarkProminic, is there a quick fix for this, then?   [This](https://stackoverflow.com/questions/63982903/apt-mark-hold-and-apt-mark-unhold-with-ansible-modules) might work to exclude the ansible package.

**Comment by MarkProminic** [comment]
Yes, change the box version as I mentioned to any version higher that 0.0.6

"If you change the vagrant box template to anything higher than 0.0.6"

**Comment by MarkProminic** [comment]
At this time, and the number of times this has happened before is low, Just using the latest box version will fix the issue

**Comment by piotrzarzycki21** [comment]
Box has been updates for 0.1.22 -> https://github.com/Moonshine-IDE/Super.Human.Installer/commit/3c8692ea5788c3d62f91aa6b2b4c172fa42e4626

@JoelProminic Can we close it ?

**Comment by piotrzarzycki21** [comment]
I'm not sure if this is fixed or not. I have made some improvements for 0.1.22 Hosts.rb file based on issue above. Test for new server and let me know. I'm moving this to next release.

**Comment by piotrzarzycki21** [comment]
Tested on Windows 11 - Worked
Tested on Mac OS Monterey 12.7.6 - Worked
